compiler/rustc_target/src/spec/mod.rs

```diff
@@ -2169,7 +2288,7 @@ impl TargetOptions {
     }
 
     fn update_to_cli(&mut self) {
-        self.linker_flavor_json = self.linker_flavor.to_cli();
+        self.linker_flavor_json = self.linker_flavor.to_cli_counterpart();
```

Comment:
We could. I don't believe it would change much if anything except serialized json would use modern names instead of old ones (and thus if used by a target spec, have both flavors in the json output). The normalization done during deserialization should mean there wouldn't be functional changes, if I understand correctly.

I'll push a commit with that change, and update the custom target spec in the end-to-end test to use the modern flavors here as well.

Reply:
Done